### PR TITLE
#1615 fix

### LINKED ISSUE #1615: `device=self.device` missing from lots of places in tensor.py
See: https://github.com/tinygrad/tinygrad/blob/4f459841bcfc57bf4f328b45553212c3bd0b6fdb/tinygrad/tensor.py#L338
And: https://github.com/tinygrad/tinygrad/blob/4f459841bcfc57bf4f328b45553212c3bd0b6fdb/tinygrad/tensor.py#L334

Should be something like: https://github.com/tinygrad/tinygrad/blob/4f459841bcfc57bf4f328b45553212c3bd0b6fdb/tinygrad/tensor.py#L711

Basically: https://github.com/tinygrad/tinygrad/pull/1591#discussion_r1300387689

We don't want to use the default device for this, should be on the same device that the other tensors are on.

### PR DESCRIPTION
**Pull request by geohotstan**
#1615
I also turned requires_grad off on some of them
also for sparse_categorical_crossentropy and argmax, should the arange be int32?

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/tensor.py                         6               6
------------------------------------------------------------
lines added in the tinygrad folder: 0
```

**Comment by wozeparrot** [comment]
yea, the dtypes should be fine for crossentropy and argmax, both are indexes

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def arange(start, stop=None, step=1, **kwargs):
 
   @staticmethod
   def full_like(tensor, fill_value, dtype:Optional[DType]=None, **kwargs):
-    return Tensor.full(tensor.shape, fill_value=fill_value, dtype=tensor.dtype if dtype is None else dtype, **kwargs)
+    return Tensor.full(tensor.shape, fill_value=fill_value, dtype=tensor.dtype if dtype is None else dtype, device=tensor.device, **kwargs)
```
Comment (ZenginU) [inline tinygrad/tensor.py:161]: isn't device is included in **kwargs

**Comment by geohotstan** [comment]
@ZenginU also found that [ones_like](https://pytorch.org/docs/stable/generated/torch.ones_like.html) and [zeros_like](https://pytorch.org/docs/stable/generated/torch.zeros_like.html) from torch
"Returns a tensor filled with the scalar value 1/0, with the same size as input. torch.ones_like(input) is equivalent to torch.ones(input.size(), dtype=input.dtype, layout=input.layout, **_device=input.device_**)"

So I suppose having full_like take the `device=tensor.device` is gooooooooooooood?

File: tinygrad/tensor.py (hunk 1 above)
Comment (wozeparrot) [inline tinygrad/tensor.py:161]: need to remove a possible device from the `kwargs` here

**Comment by wozeparrot** [comment]
yea thats fine

**Comment by wozeparrot** [comment]
merge conflict

**Pull request merged**